### PR TITLE
[Logs][Enhancement]: Expose logs expvar metrics through the agent status

### PR DESCRIPTION
**Pull request by gaetan-deputier**
### What does this PR do?

Give the customers more visibility to know if their logs are being processed by the agent or not

Screenshot:
image

### Motivation

[Trello](https://trello.com/c/hpKBtk6E/1023-agent-status-expose-log-related-metrics-that-are-stored-in-the-expvar-data-to-the-agent-status)

### CODE REVIEW

File: cmd/agent/gui/views/templates/generalStatus.tmpl [hunk 1]
```diff
@@ -206,6 +206,14 @@
 
         Logs Agent is not running </br>
       {{- end }}
+      {{- if .metrics }}
+
+         Metrics
+        {{ printDashes "Metrics" "=" }}
+        {{- range $metric_key, $metric_value := .metrics }}
```
Comment (ajacquemot) [inline cmd/agent/gui/views/templates/generalStatus.tmpl]: nits: `metric_name`

File: pkg/logs/status/builder.go [hunk 2]
```diff
@@ -133,3 +137,10 @@ func (b *Builder) toDictionary(c *config.LogsConfig) map[string]interface{} {
 	}
 	return dictionary
 }
+
+func (b *Builder) getStatusMetrics() map[string]int64 {
```
Comment (ajacquemot) [inline pkg/logs/status/builder.go]: nits `getMetricsStatus`

File: pkg/status/dist/templates/logsagent.tmpl [hunk 3]
```diff
@@ -10,6 +10,15 @@ Logs Agent
   Logs Agent is not running
 {{- end }}
 
+{{- if .metrics }}
+
+   Metrics
+  {{ printDashes "Metrics" "=" }}
+  {{- range $metric_key, $metric_value := .metrics }}
+    {{$metric_key}}: {{$metric_value}}
```
Comment (ajacquemot) [inline pkg/status/dist/templates/logsagent.tmpl]: same here

File: pkg/logs/status/builder.go [hunk 4]
```diff
@@ -133,3 +137,10 @@ func (b *Builder) toDictionary(c *config.LogsConfig) map[string]interface{} {
...
+	var metrics = make(map[string]int64, 2)
```
Comment (ajacquemot) [inline pkg/logs/status/builder.go]: why do we expose only two of them while there are three?

File: pkg/logs/status/builder.go [hunk 5]
```diff
@@ -133,3 +137,10 @@ func (b *Builder) toDictionary(c *config.LogsConfig) map[string]interface{} {
...
+	metrics["LogsProcessed"] = b.logsExpVars.Get("LogsProcessed").(*expvar.Int).Value()
```
Comment (ajacquemot) [inline pkg/logs/status/builder.go]: this is too manual, if we had a new metric, it won't be added automatically to the status.

**Review by ajacquemot — COMMENTED** [review]
Few nits and two concerns

File: pkg/logs/status/builder.go (hunk 5 above)
Comment (gaetan-deputier) [reply pkg/logs/status/builder.go]: Thanks for the review, I'll address the nits right away.

I did not iterate over the logsExpVars to respect the two metrics from the card: https://trello.com/c/hpKBtk6E/1023-agent-status-expose-log-related-metrics-that-are-stored-in-the-expvar-data-to-the-agent-status

I would need Nils to confirm but my guess would be that LogsDecoded makes less sense for customers. Now, I could add all the expvars and just iterate instead.

Comment (ajacquemot) [reply pkg/logs/status/builder.go]: Should we update the metrics though ? I started to work on it a couple of times ago https://github.com/DataDog/datadog-agent/pull/2504/files#diff-eb87589455476ecd4da9932fefcee02dR29

Comment (gaetan-deputier) [reply pkg/logs/status/builder.go]: Maybe for 6.13 if we get some feedback around those first metrics?
Let's chat why the previous PR did not go through.

**Review by ajacquemot — DISMISSED** [review]
We could have built a wrapper on top of the `expvar.Map` data structure to expose all keys to iterate on all key-values from the status so that it always contains all the metrics but we can do it later, worth adding a TODO.

**Review by arbll — APPROVED** [review]

**Review by ajacquemot — APPROVED** [review]

**Pull request merged**